src/GamingDevice.js

```diff
@@ -0,0 +1,77 @@
+let gamingDeviceId = 0;
+const defaultNumOfButtons = 18;
+
+import { GamingDeviceButton } from "./GamingDeviceButton.js";
+
+export class GamingDevice extends HTMLDivElement {
+  constructor(numOfButtons = defaultNumOfButtons) {
+    super();
+
+    const id = gamingDeviceId++;
+
+    // setup the gamepad's front end
+    this.innerHTML = `
+      <style>
+        .gaming-device {
+          border: 1px solid grey;
+          display: inline-block;
+          margin: 5px;
+        }
+        .connected {
+          background-color: lightblue;
+        }
+      </style>
+      Hello gamepad ${id}!
+    `;
+    this.classList.add("gaming-device");
+
+    // set default attributes
+    this._id = id;
+    this._connected = false;
+    this._buttons = gamingDeviceButtons();
+
+    addButtonsToButtonDiv(this);
+
+    function gamingDeviceButtons() {
```

Comment:
Nit: when naming functions, remember to use verbs instead of nouns... so like: 

```suggestion
    function createDeviceButtons() {
```

However, this can probably just be inlined, as this function is only called once ever in our code. Instead, just add a code comment if it's not obvious what the code i doing.